### PR TITLE
[Card] Fix subdued section spacing issues

### PR DESCRIPTION
**Pull request by solonaarmstrong**
### WHY are these changes introduced?

Resolves https://github.com/Shopify/polaris-react/issues/954

The first fix was here: https://github.com/Shopify/polaris-react/pull/962 but it created too much spacing with non-subdued sections and the header. This fix targets only the subdued sections.

### WHAT is this pull request doing?

Add spacing and a border between header and subdued section & footer and subdued section.

### Before

screen shot 2019-02-21 at 10 26 59 am
screen shot 2019-02-21 at 10 27 12 am
screen shot 2019-02-21 at 10 27 20 am

### After

screen shot 2019-02-21 at 10 27 38 am
screen shot 2019-02-21 at 10 27 45 am
screen shot 2019-02-21 at 10 27 49 am

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import * as React from 'react';
import {Page, Card, List, DisplayText, TextContainer} from '../src';

interface State {}

export default class Playground extends React.Component<{}, State> {
  render() {
    return (
      <Page title="">
        <TextContainer>
          <DisplayText size="medium">Card.Section</DisplayText>
          <Card>
            <Card.Section title="Deactivated reports" subdued>
              <List>
                <List.Item>Payouts</List.Item>
                <List.Item>Total Sales By Channel</List.Item>
              </List>
            </Card.Section>
          </Card>
        </TextContainer>
        <br />
        <hr />
        <br />
        <TextContainer>
          <DisplayText size="medium">
            Footer actions + subdued Card.Section
          </DisplayText>
          <Card
            secondaryFooterAction={{content: 'Dismiss'}}
            primaryFooterAction={{content: 'Export Report'}}
          >
            <Card.Section title="Deactivated reports" subdued>
              <List>
                <List.Item>Payouts</List.Item>
                <List.Item>Total Sales By Channel</List.Item>
              </List>
            </Card.Section>
          </Card>
        </TextContainer>
        <br />
        <TextContainer>
          <DisplayText size="medium">Footer actions + Card.Section</DisplayText>
          <Card
            secondaryFooterAction={{content: 'Dismiss'}}
            primaryFooterAction={{content: 'Export Report'}}
          >
            <Card.Section title="Deactivated reports">
              <List>
                <List.Item>Payouts</List.Item>
                <List.Item>Total Sales By Channel</List.Item>
              </List>
            </Card.Section>
          </Card>
        </TextContainer>
        <br />
        <hr />
        <br />
        <TextContainer>
          <DisplayText size="medium">
            Card.Header + subdued Card.Section with title
          </DisplayText>
          <Card>
            <Card.Header
              actions={[
                {
                  content: 'Preview',
                },
              ]}
              title="Staff accounts"
            />
            <Card.Section title="Deactivated reports" subdued>
              <List>
                <List.Item>Payouts</List.Item>
                <List.Item>Total Sales By Channel</List.Item>
              </List>
            </Card.Section>
          </Card>
        </TextContainer>
        <br />
        <TextContainer>
          <DisplayText size="medium">
            Card.Header + Card.Section with title
          </DisplayText>
          <Card>
            <Card.Header
              actions={[
                {
                  content: 'Preview',
                },
              ]}
              title="Staff accounts"
            />
            <Card.Section title="Deactivated reports">
              <List>
                <List.Item>Payouts</List.Item>
                <List.Item>Total Sales By Channel</List.Item>
              </List>
            </Card.Section>
          </Card>
        </TextContainer>
        <br />
        <TextContainer>
          <DisplayText size="medium">
            Card.Header + Card.Section without title
          </DisplayText>
          <Card>
            <Card.Header
              actions={[
                {
                  content: 'Preview',
                },
              ]}
              title="Staff accounts"
            />
            <Card.Section>
              <List>
                <List.Item>Payouts</List.Item>
                <List.Item>Total Sales By Channel</List.Item>
              </List>
            </Card.Section>
          </Card>
        </TextContainer>
        <br />
        <hr />
        <br />
        <TextContainer>
          <DisplayText size="medium">
            Header actions + subdued Card.Section with title
          </DisplayText>
          <Card title="Title" actions={[{content: 'Add variant'}]}>
            <Card.Section title="Deactivated reports" subdued>
              <List>
                <List.Item>Payouts</List.Item>
                <List.Item>Total Sales By Channel</List.Item>
              </List>
            </Card.Section>
          </Card>
        </TextContainer>
        <br />
        <TextContainer>
          <DisplayText size="medium">
            Header actions + Card.Section with title
          </DisplayText>
          <Card title="Title" actions={[{content: 'Add variant'}]}>
            <Card.Section title="Deactivated reports">
              <List>
                <List.Item>Payouts</List.Item>
                <List.Item>Total Sales By Channel</List.Item>
              </List>
            </Card.Section>
          </Card>
        </TextContainer>
        <br />
        <TextContainer>
          <DisplayText size="medium">
            Header actions + Card.Section without title
          </DisplayText>
          <Card title="Title" actions={[{content: 'Add variant'}]}>
            <Card.Section>
              <List>
                <List.Item>Payouts</List.Item>
                <List.Item>Total Sales By Channel</List.Item>
              </List>
            </Card.Section>
          </Card>
        </TextContainer>
        <br />
        <hr />
        <br />
        <TextContainer>
          <DisplayText size="medium">
            Multiple subdued Card.Sections
          </DisplayText>
          <Card
            title="Title"
            actions={[{content: 'Add variant'}]}
            secondaryFooterAction={{content: 'Dismiss'}}
            primaryFooterAction={{content: 'Export Report'}}
          >
            <Card.Section>
              <List>
                <List.Item>Payouts</List.Item>
                <List.Item>Total Sales By Channel</List.Item>
              </List>
            </Card.Section>
            <Card.Section subdued>
              <List>
                <List.Item>Payouts</List.Item>
                <List.Item>Total Sales By Channel</List.Item>
              </List>
            </Card.Section>
            <Card.Section subdued>
              <List>
                <List.Item>Payouts</List.Item>
                <List.Item>Total Sales By Channel</List.Item>
              </List>
            </Card.Section>
            <Card.Section>
              <List>
                <List.Item>Payouts</List.Item>
                <List.Item>Total Sales By Channel</List.Item>
              </List>
            </Card.Section>
            <Card.Section subdued>
              <List>
                <List.Item>Payouts</List.Item>
                <List.Item>Total Sales By Channel</List.Item>
              </List>
            </Card.Section>
          </Card>
        </TextContainer>
      </Page>
    );
  }
}

```

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)

### CODE REVIEW

File: src/components/Card/Card.scss [hunk 1]
```diff
@@ -22,6 +22,11 @@
   @include page-content-when-not-fully-condensed {
     padding: spacing(loose) spacing(loose) 0;
   }
+
+  + .Section-subdued {
+    border-top: border-width() solid color('sky');
+    margin-top: spacing(loose);
+  }
```
Comment (ry5n) [inline src/components/Card/Card.scss]: This is styling a subdued section, should these rules be nested under the `.Section` selector instead?

```scss
.Section {
   ...

   .Header + & {
     ...
  }
}

File: src/components/Card/Card.scss [hunk 2]
```diff
@@ -42,6 +47,11 @@
 
 .Section-subdued {
   background-color: color('sky', 'lighter');
+
+  + .Footer {
+    border-top: border-width() solid color('sky');
+    padding: spacing(loose) spacing(loose) spacing(loose);
+  }
```
Comment (ry5n) [inline src/components/Card/Card.scss]: Same here, should this be nested under `.Footer`?

**Comment by ry5n** [comment]
(Haven’t had a chance to tophat yet, which is the only reason I didn’t approve.)

File: src/components/Card/Card.scss (hunk 1 above)
Comment (solonaarmstrong) [reply src/components/Card/Card.scss]: That was fast! Thank you.

The `>` is the key here. What it's doing is targeting the subdued section only if it's the next sibling of subdued. Moving it out wouldn't achieve that.

Comment (solonaarmstrong) [reply src/components/Card/Card.scss]: I'm sorry, Ryan. I tried your suggestion and you're absolutely right. 👍

Comment (ry5n) [reply src/components/Card/Card.scss]: I do have mixed feelings about doing it this way. Nesting with `+ &` is hard to understand. By contrast, the CSS that it compiles to is plain as day: `.Header + .Section {…}`.

Separately, it’s really helpful to have all the styles for each element (in this case, all the styles for `.Section`) in one place in the file.

My personal preference would be to have both, by avoiding the `+ &` contortions but keeping the styles together:

```scss
.Section {
  // All the basic styles for the section
}

.Header + .Section {
  // Specific styles for this case right below
}
``` 

However, I think our in-house Sass style prefers nesting, so we need to do the `+ &` thing unless we want to open a discussion about code style.

File: src/components/Card/Card.scss (hunk 2 above)
Comment (ry5n) [reply src/components/Card/Card.scss]: This is resolved

File: src/components/Card/Card.scss (hunk 1 above)
Comment (solonaarmstrong) [reply src/components/Card/Card.scss]: Yes, I think the linter complains if you repeat a class.

File: src/components/Card/Card.scss [hunk 3]
```diff
@@ -66,4 +71,9 @@
   @include page-content-when-not-fully-condensed {
     padding: 0 spacing(loose) spacing(loose);
   }
+
+  .Section-subdued + & {
+    border-top: border-width() solid color('sky');
+    padding: spacing(loose) spacing(loose) spacing(loose);
```
[the remark this answers is not in the document]
Comment (solonaarmstrong) [reply src/components/Card/Card.scss]: Oh, you're right, the left padding doesn't matter.

**Comment by chloerice** [comment]
I've been having a hard time getting `web` to run 😣Gonna switch gears and compare master to these changes, brb...

**Pull request merged**